elixir/lib/extism/plugin.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Extism.Plugin do
+  defstruct [
+    # The actual NIF Resource.
+    resource: nil,
+    # Normally the compiler will happily do stuff like inlining the
+    # resource in attributes. This will convert the resource into an
+    # empty binary with no warning. This will make that harder to
+    # accidentaly do.
+    # It also serves as a handy way to tell plugins apart.
+    reference: nil
+  ]
+
+  def wrap_resource(resource) do
+    %__MODULE__{
+      resource: resource,
```

Comment:
`resource` is just the PluginIndex here. The elixir user sees the `%Plugin` type as the reference. We may not need the `reference` property here I just cargo culted it.